tests/Bridge/Doctrine/Orm/Filter/BooleanFilterTest.php

```diff
@@ -0,0 +1,246 @@
+
+    /**
+     * @var ManagerRegistry
+     */
+    private $managerRegistry;
+
+    /**
+     * @var EntityRepository
+     */
+    private $repository;
+
+    /**
+     * @var string
+     */
+    protected $resourceClass;
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function setUp()
+    {
+        self::bootKernel();
+        $manager = DoctrineTestHelper::createTestEntityManager();
+        $this->managerRegistry = self::$kernel->getContainer()->get('doctrine');
+        $this->repository = $manager->getRepository(Dummy::class);
+        $this->resourceClass = Dummy::class;
+    }
+
+    /**
+     * @dataProvider filterProvider
+     */
+    public function testApply(array $filterParameters, array $query, $expected)
+    {
+        $request = Request::create('/api/dummies', 'GET', $query);
+        $requestStack = new RequestStack();
+        $requestStack->push($request);
+        $queryBuilder = $this->repository->createQueryBuilder('o');
+        $filter = new BooleanFilter(
+            $this->managerRegistry,
+            $requestStack,
+            $filterParameters['properties']
+        );
+
+        $uniqid = $this->getFunctionMock('ApiPlatform\Core\Bridge\Doctrine\Orm\Util', 'uniqid');
+        $uniqid->expects($this->any())->willReturn('123456abcdefg');
+
+        $filter->apply($queryBuilder, $this->resourceClass);
+        $actual = strtolower($queryBuilder->getQuery()->getDQL());
+        $expected = strtolower($expected);
+
+        $this->assertEquals(
+            $expected,
+            $actual,
+            sprintf('Expected `%s` for this `%s %s` request', $expected, 'GET', $request->getUri())
+        );
+    }
+
+    public function testGetDescription()
+    {
+        $filter = new BooleanFilter($this->managerRegistry, new RequestStack(),  [
+            'id' => null,
+            'name' => null,
+            'foo' => null,
+            'dummyBoolean' => null,
+        ]);
+        $this->assertEquals([
+            'dummyBoolean' => [
+                'property' => 'dummyBoolean',
+                'type' => 'boolean',
+                'required' => false,
+            ],
+        ], $filter->getDescription($this->resourceClass));
+    }
+
+    public function testGetDescriptionDefaultFields()
+    {
+        $filter = new BooleanFilter($this->managerRegistry, new RequestStack());
+        $this->assertEquals([
+            'dummyBoolean' => [
+                'property' => 'dummyBoolean',
+                'type' => 'boolean',
+                'required' => false,
+            ],
+        ], $filter->getDescription($this->resourceClass));
+    }
+
+    /**
+     * Providers 3 parameters:
+     *  - filter parameters.
+     *  - properties to test. Keys are the property name. If the value is true, the filter should work on the property,
+     *    otherwise not.
+     *  - expected DQL query.
+     *
+     * @return array
+     */
+    public function filterProvider()
+    {
+        return [
```

Comment:
But are the keys passed? When using a data provider, you get the content of the array values, not the keys. Or did I miss something?